packages/block-editor/src/components/caption/index.native.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { RichText } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+const Caption = ( { accessible, accessibilityLabel, onBlur, onChange, onFocus, isParentSelected, isSelected, text } ) => {
+	// We still want to render the caption so that the soft keyboard is not forced to close on Android
+	const shouldCaptionDisplay = ! RichText.isEmpty( text ) > 0 || isParentSelected;
+
+	return (
+		<View style={ {	padding: 12, flex: 1, display: shouldCaptionDisplay ? 'flex' : 'none' } }
+			accessible={ accessible }
+			accessibilityLabel={ accessibilityLabel }
+			accessibilityRole={ 'button' }
+		>
+			<RichText
+				rootTagsToEliminate={ [ 'p' ] }
+				placeholder={ __( 'Write caption…' ) }
+				value={ text }
```

Comment:
Since we're in the block editor package, I think we could easily "connect" this component to the store to get/update the `caption` attribute. It could use a `clientId` as a props instead and we would be removing `onChange` and `text`.

Reply:
That's a really interesting idea! Thanks for sharing it! I'm not sure I know exactly how to do what you suggest 😁, but I took a shot at it and I have updated this PR. It might be easier for you to just look at [the specific commit](https://github.com/WordPress/gutenberg/pull/16825/commits/28a11442a105779d8e8dee35e53a9b339cea0891) that makes those changes. I think I like this a lot better. The one concern I have is that the `Caption` component now depends on being passed a `clientId` that will get it a an attribute object with a `caption` property. I think this is ok, but I don't love it. What do you think about that and about this approach in general? No worries if this isn't what you had in mind or is a bad idea for some reason. I can easily revert the change.

:wave: @etoledom : if you want to give the latest commit on this PR a look, I'd be interested in hearing any thoughts you might have too!